### PR TITLE
Linux action

### PR DESCRIPTION
**Pull request by bam241**
some test around GitHub action

### CODE REVIEW

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+          echo "CURRENT=$(pwd)" >> $GITHUB_ENV
+          echo "PLUGIN_ABS_PATH=" >> $GITHUB_ENV
+          echo "SCRIPTPATH=$GITHUB_WORKSPACE" >> $GITHUB_ENV
+          echo "PLUGIN_ABS_PATH=$GITHUB_WORKSPACE/.." >> $GITHUB_ENV
+
+      - name: Apt dependencies
+        shell: bash
+        run: |
+          sudo apt-get update -y
+          sudo apt-get install -y g++ libtool libeigen3-dev libhdf5-dev patchelf git cmake
```
Comment (gonuke) [inline .github/workflows/linux.yml]: Do we need `libtool`?

File: script/linux_share_build.sh [hunk 2]
```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+
+
+function install_prerequise() {
+    TZ=America/Chicago
+    sudo ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
+    sudo apt-get update -y
+    sudo apt-get install -y g++ libtool libeigen3-dev libhdf5-dev patchelf git cmake
```
Comment (gonuke) [inline script/linux_share_build.sh]: Same question about `libtool`

File: script/build_plug_linux.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```
Comment (gonuke) [inline script/build_plug_linux.sh]: Maybe some documentation on the arguments?

**Review by gonuke — COMMENTED** [review]
I think this looks ok.  It's a lot of building of MOAB & DAGMC - can we eventually provide our own docker image?

File: .github/workflows/linux.yml [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    branches:
+      - develop
+      - master
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  main:
+    runs-on: ubuntu-18.04
```
Comment (pshriwise) [inline .github/workflows/linux.yml:20]: If we can build this for multiple Ubuntu versions, we may as well do it. Unless we're confident that the plugin from 18.04 works for 20.04 of course.

File: .github/workflows/linux.yml [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    name: 'Trelis Svalinn Plugin 17.1 Build'
```
Comment (pshriwise) [inline .github/workflows/linux.yml]: Should this name have Linux, Ubuntu, and/or the OS version in it for clarity?

**Review by pshriwise — COMMENTED** [review]
A couple of comments, but also a general question here. We have scripts for building the plugin for Linux and most of the lines here are copied from there. Can those scripts be refactored so that we can source the `linux_build_share.sh` file and call the functions from there for each step in this file? It would improve the maintenance outlook here I think.

File: .github/workflows/linux.yml (hunk 4 above)
Comment (bam241) [reply .github/workflows/linux.yml:20]: I think that is ok, as we ship most of the lib we depend on. (mainly hdf5...)

Comment (pshriwise) [reply .github/workflows/linux.yml:20]: So the plugin built with Ubuntu 18.04 will work on 20.04?

**Comment by bam241** [comment]
closing to be reopened in some other form

**Pull request closed without merging**